### ISSUE TITLE
BulkInsertOrUpdateOrDelete does nothing

### ISSUE THREAD

**Issue by rcausier** [borisdj/EFCore.BulkExtensions#878]
Hello, I use your BulkInsertOrUpdateOrDelete function at many times and it workd fine ! Thanks a lot.

But in a specific case (and table), it does nothing.
I have SQL Server 2016.

I put the profiler to see the SQL requets.

When it does not work, the only SQL request I can see is : SELECT TOP 0 ... INTO #temporaryTable ....
The table is created but no more SQL requests are done (no MERGE for example).

My code for all BULKs (the same code for tables which works and tables which don't work) :

        webDbContext.BulkSynchronize(appliedEvent.Product.Shops, e => e.ProductId == appliedEvent.Product.Id);

        public void BulkSynchronize<T>(IList<T> Items, Expression<Func<T, bool>> filter) where T : class
        {
            var bulkConfig = new BulkConfig { UseTempDB = true };
            bulkConfig.SetSynchronizeFilter<T>(filter);
            using var transaction = Database.BeginTransaction();
            this.BulkInsertOrUpdateOrDelete(Items, bulkConfig);
            transaction.Commit();
        }

The table :

CREATE TABLE [dbo].[ShopProduct](
	[ProductId] [int] NOT NULL,
	[ShopId] [int] NOT NULL,
	[ShipDiscount] [money] NULL,
	[WeightClass] [float] NULL,
	[Visibility] [int] NOT NULL,
	[VisibleInAccessoriesOnly] [bit] NOT NULL,
	[WeightClassRelay] [float] NULL,
	[WarehouseId] [int] NULL,
	[HiddenForcedBySage] [bit] NOT NULL,
 CONSTRAINT [PK_ProdShopSubs] PRIMARY KEY CLUSTERED 
(
	[ProductId] ASC,
	[ShopId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO

The EntityConfiguration :

        public override void Configure(EntityTypeBuilder<ProductByShopModel> builder)
        {
            builder.CheckForNullReference();

            builder.ToTable("ShopProduct");

            builder.HasKey(x => new { x.ProductId, x.ShopId });

            builder.Property(x => x.ProductId).HasColumnName("ProductId");
            builder.Property(x => x.ShopId).HasColumnName("ShopId");
            builder.Property(x => x.ShipDiscount).HasColumnName("ShipDiscount").HasColumnType("money");
            builder.Property(x => x.WeightClass).HasColumnName("WeightClass");
            builder.Property(x => x.WeightClassRelay).HasColumnName("WeightClassRelay").HasColumnType("float");
            builder.Property(x => x.Visibility).HasColumnName("Visibility");
            builder.Property(x => x.VisibleInAccessoriesOnly).HasColumnName("VisibleInAccessoriesOnly");
            builder.Property(x => x.WarehouseId).HasColumnName("WarehouseId");
            builder.Property(x => x.HiddenForcedBySage).HasColumnName("HiddenForcedBySage");
        }

Do you have any idea to fix this ? or to log what's wrong ?

Thanks a lot

**Comment by borisdj** [comment]
Add the source of lib. into the project and try debugging it.

**Comment by rcausier** [comment]
System.IO.FileNotFoundException : 'Could not load file or assembly 'Npgsql.EntityFrameworkCore.PostgreSQL, Version=6.0.5.0, Culture=neutral, PublicKeyToken=5d8b90d52f46fda7'. Le fichier spécifié est introuvable.'

My databse is SQL Server but the TableInfo.loaddata(context, type, entities, isDeleteOperation) line fails on finding a postgresql dll ???

thanks for help

**Comment by rcausier** [comment]
OK I found and fixed the issue (debugging your code was a good solution). 
After fixing the issue, I realized that errors were bad catched in our program, now it's resolved and I can see error messages.
The error message was : 

System.InvalidOperationException : 'The configured execution strategy 'SqlServerRetryingExecutionStrategy' 
does not support user-initiated transactions. 

Use the execution strategy returned by 'DbContext.Database.CreateExecutionStrategy()' to execute all the operations 
in the transaction as a retriable unit.'

I fixed my code like this: 

public void BulkSynchronize<T>(IList<T> Items, Expression<Func<T, bool>> filter) where T : class
{
	var bulkConfig = new BulkConfig { UseTempDB = true };
	bulkConfig.SetSynchronizeFilter<T>(filter);
	var strategy = Database.CreateExecutionStrategy();
	strategy.Execute(() =>
	{
		using var transaction = Database.BeginTransaction();
		this.BulkInsertOrUpdateOrDelete(Items, bulkConfig);
		transaction.Commit();
	});
}

Hope it will help someone else.
